Thank you @saiaunghlyanhtet !

Looks like it panics when running on a cluster with `enable-ipv6: false`. It might be better to use [ForEachIPFamily](https://github.com/cilium/cilium/blob/8b7410510ac5c4f3631187428d69bd9622f8071f/cilium-cli/connectivity/check/test.go#L845) when running tests for each IP family, depending on the cluster configuration.

There's an issue where the agent crashes when deleting a CLRP while IPv6 is disabled. I can take care of this one.
https://github.com/cilium/cilium/issues/38570

```
[=] [cilium-test-1] Test [local-redirect-policy] [101/114]
  ℹ️  📜 Applying CiliumLocalRedirectPolicy 'lrp-address-matcher-v4' to namespace 'cilium-test-1' on cluster kind-kind..
  ℹ️  📜 Applying CiliumLocalRedirectPolicy 'lrp-address-matcher-v6' to namespace 'cilium-test-1' on cluster kind-kind..
  ℹ️  📜 Applying CiliumNetworkPolicy 'client-egress-to-cidr-lrp-deny' to namespace 'cilium-test-1' on cluster kind-kind..
  ℹ️  📜 Applying CiliumLocalRedirectPolicy 'lrp-address-matcher-skip-redirect-from-backend-v4' to namespace 'cilium-test-1' on cluster kind-kind..
  ℹ️  📜 Applying CiliumLocalRedirectPolicy 'lrp-address-matcher-skip-redirect-from-backend-v6' to namespace 'cilium-test-1' on cluster kind-kind..
  [-] Scenario [local-redirect-policy/lrp]
  [.] Action [local-redirect-policy/lrp/curl-0-ipv4: cilium-test-1/lrp-client-55f54dbf99-7t8lk (10.244.1.179) -> 169.254.169.254:80/TCP (169.254.169.254:80)]
.  [.] Action [local-redirect-policy/lrp/curl-0-ipv4: cilium-test-1/lrp-backend-5dc456df96-9hqc8 (10.244.1.238) -> 169.254.169.254:80/TCP (169.254.169.254:80)]
.  ℹ️  📜 Deleting CiliumLocalRedirectPolicy 'lrp-address-matcher-v4' in namespace 'cilium-test-1' on cluster kind-kind..
  ℹ️  📜 Deleting CiliumLocalRedirectPolicy 'lrp-address-matcher-v6' in namespace 'cilium-test-1' on cluster kind-kind..
  ℹ️  📜 Deleting CiliumNetworkPolicy 'client-egress-to-cidr-lrp-deny' in namespace 'cilium-test-1' on cluster kind-kind..
  ℹ️  📜 Deleting CiliumLocalRedirectPolicy 'lrp-address-matcher-skip-redirect-from-backend-v4' in namespace 'cilium-test-1' on cluster kind-kind..
  ℹ️  📜 Deleting CiliumLocalRedirectPolicy 'lrp-address-matcher-skip-redirect-from-backend-v6' in namespace 'cilium-test-1' on cluster kind-kind..
panic: runtime error: index out of range [1] with length 1

goroutine 554 [running]:
github.com/cilium/cilium/cilium-cli/connectivity/tests.lrp.runTestsForIPFamily({{{0xa0bbc2a, 0x24}}, 0x0}, {0x8c210b8, 0xc00087d480}, 0xc00123ac60, {0xc000a370e8, 0x1, 0x1?}, 0x2)
	/home/yusuke/go/src/github.com/cilium/cilium/cilium-cli/connectivity/tests/lrp.go:128 +0xda5
github.com/cilium/cilium/cilium-cli/connectivity/tests.lrp.Run({{{0xa0bbc2a?, 0x5ccd0a0?}}, 0x60?}, {0x8c210b8, 0xc00087d480}, 0xc00123ac60)
	/home/yusuke/go/src/github.com/cilium/cilium/cilium-cli/connectivity/tests/lrp.go:97 +0x608
github.com/cilium/cilium/cilium-cli/connectivity/check.(*Test).Run(0xc00123ac60, {0x8c210b8, 0xc00087d480}, 0x65)
	/home/yusuke/go/src/github.com/cilium/cilium/cilium-cli/connectivity/check/test.go:382 +0x664
github.com/cilium/cilium/cilium-cli/connectivity/check.(*ConnectivityTest).Run.func1()
	/home/yusuke/go/src/github.com/cilium/cilium/cilium-cli/connectivity/check/context.go:449 +0x99
created by github.com/cilium/cilium/cilium-cli/connectivity/check.(*ConnectivityTest).Run in goroutine 494
	/home/yusuke/go/src/github.com/cilium/cilium/cilium-cli/connectivity/check/context.go:443 +0x91
```

--- a/cilium-cli/connectivity/builder/local_redirect_policy.go
+++ b/cilium-cli/connectivity/builder/local_redirect_policy.go
@@ -22,8 +22,11 @@ var (
 type localRedirectPolicy struct{}
 
 func (t localRedirectPolicy) build(ct *check.ConnectivityTest, _ map[string]string) {
-	lrpFrontendIP := "169.254.169.254"
-	lrpFrontendIPSkipRedirect := "169.254.169.255"
+	lrpFrontendIPV4 := "169.254.169.254"
+	lrpFrontendIPV6 := "fd00::169:254:169:254"
+	lrpFrontendIPSkipRedirectV4 := "169.254.169.255"
+	lrpFrontendIPSkipRedirectV6 := "fd00::169:254:169:255"
+
 	newTest("local-redirect-policy", ct).
 		WithCondition(func() bool {
 			if versioncheck.MustCompile(">=1.16.0")(ct.CiliumVersion) {
@@ -35,15 +38,27 @@ func (t localRedirectPolicy) build(ct *check.ConnectivityTest, _ map[string]stri
 		}).
 		WithCiliumLocalRedirectPolicy(check.CiliumLocalRedirectPolicyParams{
 			Policy:                  localRedirectPolicyYAML,
-			Name:                    "lrp-address-matcher",
-			FrontendIP:              lrpFrontendIP,
+			Name:                    "lrp-address-matcher-v4",
+			FrontendIP:              lrpFrontendIPV4,
+			SkipRedirectFromBackend: false,
+		}).
+		WithCiliumLocalRedirectPolicy(check.CiliumLocalRedirectPolicyParams{
+			Policy:                  localRedirectPolicyYAML,
+			Name:                    "lrp-address-matcher-v6",
+			FrontendIP:              lrpFrontendIPV6,
 			SkipRedirectFromBackend: false,
 		}).
 		WithCiliumPolicy(localRedirectPolicyFrontendDenyYAML).
 		WithCiliumLocalRedirectPolicy(check.CiliumLocalRedirectPolicyParams{
 			Policy:                  localRedirectPolicyYAML,
-			Name:                    "lrp-address-matcher-skip-redirect-from-backend",
-			FrontendIP:              lrpFrontendIPSkipRedirect,
+			Name:                    "lrp-address-matcher-skip-redirect-from-backend-v4",
+			FrontendIP:              lrpFrontendIPSkipRedirectV4,
+			SkipRedirectFromBackend: true,
+		}).
+		WithCiliumLocalRedirectPolicy(check.CiliumLocalRedirectPolicyParams{
+			Policy:                  localRedirectPolicyYAML,
+			Name:                    "lrp-address-matcher-skip-redirect-from-backend-v6",
+			FrontendIP:              lrpFrontendIPSkipRedirectV6,
 			SkipRedirectFromBackend: true,
 		}).
 		WithFeatureRequirements(features.RequireEnabled(features.LocalRedirectPolicy)).
@@ -53,9 +68,13 @@ func (t localRedirectPolicy) build(ct *check.ConnectivityTest, _ map[string]stri
 		).
 		WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
 			if a.Scenario().Name() == "lrp-skip-redirect-from-backend" {
-				if a.Source().HasLabel("lrp", "backend") &&
-					a.Destination().Address(features.IPFamilyV4) == lrpFrontendIPSkipRedirect {
-					return check.ResultPolicyDenyEgressDrop, check.ResultNone
+				if a.Source().HasLabel("lrp", "backend") {
+					if a.Destination().Address(features.IPFamilyV4) == lrpFrontendIPSkipRedirectV4 {
+						return check.ResultPolicyDenyEgressDrop, check.ResultNone
+					}
+					if a.Destination().Address(features.IPFamilyV6) == lrpFrontendIPSkipRedirectV6 {
+						return check.ResultPolicyDenyEgressDrop, check.ResultNone
+					}
 				}
 				return check.ResultOK, check.ResultNone
 			}

--- a/cilium-cli/connectivity/builder/manifests/client-egress-to-cidr-lrp-frontend-deny.yaml
+++ b/cilium-cli/connectivity/builder/manifests/client-egress-to-cidr-lrp-frontend-deny.yaml
@@ -13,3 +13,4 @@ spec:
   egressDeny:
   - toCIDRSet:
     - cidr: 169.254.169.255/32
+    - cidr: fd00::169:254:169:255/128

--- a/cilium-cli/connectivity/tests/lrp.go
+++ b/cilium-cli/connectivity/tests/lrp.go
@@ -44,8 +44,14 @@ func (s lrp) Name() string {
 	return "lrp"
 }
 
+func getIPFamily(ip string) features.IPFamily {
+	if strings.Contains(ip, ":") {
+		return features.IPFamilyV6
+	}
+	return features.IPFamilyV4
+}
+
 func (s lrp) Run(ctx context.Context, t *check.Test) {
-	ct := t.Context()
 	policies := make([]*v2.CiliumLocalRedirectPolicy, 0, len(t.CiliumLocalRedirectPolicies()))
 
 	for _, policy := range t.CiliumLocalRedirectPolicies() {
@@ -54,22 +60,79 @@ func (s lrp) Run(ctx context.Context, t *check.Test) {
 			continue
 		}
 		policies = append(policies, policy)
+	}
+
+	t.ForEachIPFamily(func(ipFamily features.IPFamily) {
+		// Filter policies by IP family
+		var familyPolicies []*v2.CiliumLocalRedirectPolicy
+		for _, policy := range policies {
+			frontendIP := policy.Spec.RedirectFrontend.AddressMatcher.IP
+			if getIPFamily(frontendIP) == ipFamily {
+				familyPolicies = append(familyPolicies, policy)
+			}
+		}
+
+		if len(familyPolicies) == 0 {
+			return
+		}
+
+		if ipFamily == features.IPFamilyV4 {
+			s.runTestsForIPFamily(ctx, t, familyPolicies, ipFamily)
+		} else if ipFamily == features.IPFamilyV6 {
+			// Split IPv6 policies based on skipRedirectFromBackend
+			ipv6SkipTruePolicies := make([]*v2.CiliumLocalRedirectPolicy, 0)
+			ipv6SkipFalsePolicies := make([]*v2.CiliumLocalRedirectPolicy, 0)
+			for _, policy := range familyPolicies {
+				if policy.Spec.SkipRedirectFromBackend {
+					ipv6SkipTruePolicies = append(ipv6SkipTruePolicies, policy)
+				} else {
+					ipv6SkipFalsePolicies = append(ipv6SkipFalsePolicies, policy)
+				}
+			}
+
+			// Run tests for skipRedirectFromBackend=true policies regardless of SocketLB
+			if len(ipv6SkipTruePolicies) > 0 {
+				s.runTestsForIPFamily(ctx, t, ipv6SkipTruePolicies, ipFamily)
+			}
+
+			// Run tests for skipRedirectFromBackend=false policies only if SocketLB is fully functional
+			if len(ipv6SkipFalsePolicies) > 0 {
+				if !isSocketLBFull(t.Context()) {
+					t.Logf("Skipping IPv6 tests for policies with skipRedirectFromBackend=false due to SocketLB not being fully functional")
+				} else {
+					s.runTestsForIPFamily(ctx, t, ipv6SkipFalsePolicies, ipFamily)
+				}
+			}
+		}
+	})
+}
+
+func (s lrp) runTestsForIPFamily(ctx context.Context, t *check.Test, policies []*v2.CiliumLocalRedirectPolicy, ipFamily features.IPFamily) {
+	ct := t.Context()
+
+	for _, policy := range policies {
+		spec := policy.Spec
 		frontend := check.NewLRPFrontend(spec.RedirectFrontend)
-		frontendStr := net.JoinHostPort(frontend.Address(features.IPFamilyV4), fmt.Sprint(frontend.Port()))
+		frontendStr := net.JoinHostPort(frontend.Address(ipFamily), fmt.Sprint(frontend.Port()))
 		if versioncheck.MustCompile(">=1.17.0")(ct.CiliumVersion) {
 			frontendStr += fmt.Sprintf("/%s", frontend.Protocol())
 		}
+
 		lrpBackendsMap := make(map[string][]string)
 		// Check for LRP backend pods deployed on nodes in the cluster.
 		for _, pod := range t.Context().LrpBackendPods() {
 			node := pod.NodeName()
 			podIP := pod.Pod.Status.PodIP
+			if ipFamily == features.IPFamilyV6 {
+				podIP = pod.Pod.Status.PodIPs[1].IP
+			}
 			if _, ok := lrpBackendsMap[node]; !ok {
 				lrpBackendsMap[node] = []string{podIP}
 				continue
 			}
 			lrpBackendsMap[node] = append(lrpBackendsMap[node], podIP)
 		}
+
 		// Wait until the local redirect entries are plumbed in the BPF LB map
 		// on the cilium agent nodes hosting LRP backend pods.
 		WaitForLocalRedirectBPFEntries(ctx, t, frontendStr, lrpBackendsMap)
@@ -85,7 +148,7 @@ func (s lrp) Run(ctx context.Context, t *check.Test) {
 
 			i := 0
 			lf := check.NewLRPFrontend(policy.Spec.RedirectFrontend)
-			t.NewAction(s, fmt.Sprintf("curl-%d", i), &pod, lf, features.IPFamilyV4).Run(func(a *check.Action) {
+			t.NewAction(s, fmt.Sprintf("curl-%d-%s", i, ipFamily), &pod, lf, ipFamily).Run(func(a *check.Action) {
 				a.ExecInPod(ctx, a.CurlCommand(lf))
 				i++
 			})
@@ -102,7 +165,7 @@ func (s lrp) Run(ctx context.Context, t *check.Test) {
 
 			i := 0
 			lf := check.NewLRPFrontend(policy.Spec.RedirectFrontend)
-			t.NewAction(s, fmt.Sprintf("curl-%d", i), &pod, lf, features.IPFamilyV4).Run(func(a *check.Action) {
+			t.NewAction(s, fmt.Sprintf("curl-%d-%s", i, ipFamily), &pod, lf, ipFamily).Run(func(a *check.Action) {
 				a.ExecInPod(ctx, a.CurlCommand(lf))
 
 				if policy.Spec.SkipRedirectFromBackend {
@@ -113,8 +176,16 @@ func (s lrp) Run(ctx context.Context, t *check.Test) {
 				i++
 			})
 		}
-
 	}
+}
+
+func isSocketLBFull(ct *check.ConnectivityTest) bool {
+	socketLBEnabled, _ := ct.Features.MatchRequirements(features.RequireEnabled(features.KPRSocketLB))
+	if socketLBEnabled {
+		socketLBHostnsOnly, _ := ct.Features.MatchRequirements(features.RequireEnabled(features.KPRSocketLBHostnsOnly))
+		return !socketLBHostnsOnly
+	}
+	return false
 }
 
 func WaitForLocalRedirectBPFEntries(ctx context.Context, t *check.Test, frontend string, backendsMap map[string][]string) {

